### ISSUE TITLE
Foundations Module

### ISSUE THREAD

**Issue by pietercollins** [ariadne-cps/ariadne#442]
Some functionality, notably logical classes and those (like `Ball` and `Bound`) relating to topology, are currently in the `Numeric` module, but do not really belong there. I propose adding a `Foundations` module for these basic classes.

**Comment by lgeretti** [comment]
Agreed, although Foundations seems a bit too generic to me. Any more concrete names we can use?
